script.js

```diff
@@ -1,4 +1,80 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var ownchoice = input.toLowerCase();
+  var computerChoice = random_sps();
+
+  var myOutputValue = "";
+  var sps = "";
+
+  if (computerChoice == 1) {
+    sps = "scissors";
+  }
+
+  if (computerChoice == 2) {
+    sps = "paper";
+  }
+
+  if (computerChoice == 3) {
+    sps = "stone";
+  }
+
+  myOutputValue =
+    win_or_lose(ownchoice, sps) +
+    "<br> <br>" +
+    "Now you can type 'scissors', 'paper' or 'stone' to start a new game. ";
+
+  console.log("You picked", ownchoice);
+  console.log("Computer picked", sps);
+
   return myOutputValue;
 };
+
+var random_sps = function () {
+  var randomNumber = Math.random() * 3;
+  var integernumber = Math.floor(randomNumber);
+  var choices = integernumber + 1;
+  return choices;
+};
+
+var win_or_lose = function (ownchoice, sps) {
+  var msg = "Please select scissors, paper or stone.";
+  if (
+    ownchoice == sps ||
+    (ownchoice == "reversed scissors" && sps == "scissors") ||
+    (ownchoice == "reversed paper" && sps == "paper") ||
+    (ownchoice == "reversed stone" && sps == "stone")
+  ) {
+    msg =
+      "You picked " +
+      ownchoice +
+      " <br> COM picked " +
+      sps +
+      "<br> It's a draw! ";
+    return msg;
```

Comment:
You will not need a `return` keyword here, because you are `return`ing `msg` on line 79 no matter what.

The `return` keyword will be used if you want to terminate the function early, but it's not necessary because the rest of the file from line 40 onwards are just `if`-blocks.

JavaScript wouldn't execute the lines in the following `if`-blocks if the conditions are not met. This is particularly true for Scissors, Paper, Stone project, where the different conditions are very distinct. You have only 4 very separate conditions to account for: invalid input, you win, you lose, you draw. These conditions do not overlap in anyway.

Thus, in your `if`-block, you need only change the `msg` variable to store a different string for feedback for loss, win, draw, or invalid input. You do not have to `return msg;` in these `if`-blocks. We can just leave it to the final `return msg;` in line 79 to print the final value in `msg` on to the gray output box in `index.html`